rwatch/storage/storage_persist.c

```diff
@@ -0,0 +1,243 @@
+/* storage_persist.c
+ * Persist data storage api
+ * libRebbleOS
+ *
+ * Authors: Barry Carter <barry.carter@gmail.com>
+ */
+#include "rebbleos.h"
+#include "rdb.h"
+#include "storage_persist.h"
+#include "appmanager_thread.h"
+
+#define MODULE_NAME "stor"
+#define MODULE_TYPE "RWAT"
+#define LOG_LEVEL RBL_LOG_LEVEL_DEBUG //RBL_LOG_LEVEL_ERROR
+
+#define PERSIST_MAX_STORAGE_BYTES 4096
+
+typedef struct app_persist_key_t {
+    uint32_t appid;
+    uint32_t key;
+} app_persist_key;
+
+bool persist_exists(const uint32_t key)
+{
+    char buffer[1];
+    return persist_read(key, buffer, 1) == 1;
+}
+
+int persist_get_size(const uint32_t key)
+{
+    struct rdb_iter it;
+    rdb_select_result_list head;
+    list_init_head(&head);
+    
+    struct rdb_database *db = rdb_open(RDB_ID_APP_PERSIST);
+    assert(db);
+    
+    int rv = rdb_iter_start(db, &it);
+    if (!rv) {
+        rdb_close(db);
+        return E_ERROR;
+    }
+    
+    struct App *app = appmanager_get_current_app();
+    assert(app);
+    
+    app_persist_key c_key = { .appid = app->id, .key = key };    
```

Comment:
Not blocking this merge, but a TODO in the future: is app->id the correct thing here, or should it be app->uuid?  Otherwise when we upload a new version of the app (or the app gets uninstalled and reinstalled) it'd get a new id, and you'd lose the db, I think?

Reply:
yes, actually you make a really good point here. I'll probably look at changing this before we merge